.github/workflows/msys2.yml

```
@@ -111,9 +112,14 @@ jobs:
 
           if [[ '${{ matrix.COMPILER }}' == 'clang' ]]; then
             export CC=clang
+            export CC_LD=ld.lld
             export CXX=clang++
+            export CXX_LD=ld.lld
             export OBJC=clang
+            export OBJC_LD=ld.lld
             export OBJCXX=clang++
+            export OBJCXX_LD=ld.lld
+            export AR=llvm-ar
```

Maybe this?

```suggestion
            export CC_LD=lld
            export CXX=clang++
            export CXX_LD=lld
            export OBJC=clang
            export OBJC_LD=lld
            export OBJCXX=clang++
            export OBJCXX_LD=lld
            export AR=llvm-ar
```

Note that the combination clang+lld+libstdc++ is not really used in MSYS2, so there might be bugs.

---

Thanks for the help.  That's fixed one significant problem.  I'll work through the remaining ones when I get time.
I see one of them is a `test_pefile_checksum` assertion failure that I suspect is down to the previous gnu ld.bfd linker writing out some checksum field in the PE format, which llvm ld.lld may not do yet.  I'll find out for sure and potentially add case to handle different `ld_id = cc.get_linker_id()` expectations.